src/build_options.rs

```diff
@@ -158,7 +162,14 @@ impl BuildOptions {
             extra_metadata.python_source.as_deref(),
         )?;
 
-        let mut cargo_extra_args = split_extra_args(&self.cargo_extra_args)?;
+        let mut cargo_extra_args = self.cargo_extra_args.clone();
+        if cargo_extra_args.is_empty() {
+            // if not supplied on command line, try pyproject.toml
+            if let Some(args) = pyproject.and_then(|x| x.cargo_extra_args()) {
+                cargo_extra_args.push(args.to_string());
+            }
+        }
```

Comment:
I wonder how to communicate to the user when some setting was loaded from `tool.maturin.some-field` and when the cli overwrote `tool.maturin.some-field`. Maybe it a generic "loading configuration from `[tool.maturin]`" and some message for which fields the cli overwrote the `[tool.maturin]` values. Otherwise it could be rather confusing if you e.g. `git pull` a project, run `maturin build` and then get an error that only occurs because `[tool.maturin]` overwrote some logic when the defaults would have worked.

Reply:
Thanks for the review. We certainly should inform user that we used configuration from `[tool.maturin]` section.